sklearn/experimental/_column_transformer.py

```diff
@@ -0,0 +1,470 @@
+           [ 0.5,  0.5,  0. ,  1. ]])
+
+    """
+
+    def __init__(self, transformers, n_jobs=1, transformer_weights=None):
+        self.transformers = transformers
+        self.n_jobs = n_jobs
+        self.transformer_weights = transformer_weights
+
+    @property
+    def _transformers(self):
+        """
+        Internal list of transformer only containing the name and
+        transformers, dropping the columns. This is for the implementation
+        of get_params via BaseComposition._get_params which expects lists
+        of tuples of len 2.
+        """
+        return [(name, trans) for name, trans, _ in self.transformers]
+
+    @_transformers.setter
+    def _transformers(self, value):
+        self.transformers = [
+            (name, trans, col) for ((name, trans), (_, _, col))
+            in zip(value, self.transformers)]
+
+    def get_params(self, deep=True):
+        """Get parameters for this estimator.
+
+        Parameters
+        ----------
+        deep : boolean, optional
+            If True, will return the parameters for this estimator and
+            contained subobjects that are estimators.
+
+        Returns
+        -------
+        params : mapping of string to any
+            Parameter names mapped to their values.
+        """
+        return self._get_params('_transformers', deep=deep)
+
+    def set_params(self, **kwargs):
+        """Set the parameters of this estimator.
+
+        Valid parameter keys can be listed with ``get_params()``.
+
+        Returns
+        -------
+        self
+        """
+        self._set_params('_transformers', **kwargs)
+        return self
+
+    def _iter(self, X=None, skip_none=True, fitted=False):
+        """Generate (name, trans, column, weight) tuples
+        """
+        if fitted:
+            transformers = self.transformers_
+        else:
+            transformers = self.transformers
+        get_weight = (self.transformer_weights or {}).get
+        return ((name, trans, _get_column(X, column) if X is not None else X,
+                 get_weight(name))
+                for name, trans, column in transformers
+                if not skip_none or trans is not None)
+
+    def _validate_transformers(self):
+        names, transformers, _, _ = zip(*self._iter(skip_none=False))
+
+        # validate names
+        self._validate_names(names)
+
+        # validate estimators
+        for t in transformers:
+            if t is None:
+                continue
+            if (not (hasattr(t, "fit") or hasattr(t, "fit_transform")) or not
+                    hasattr(t, "transform")):
+                raise TypeError("All estimators should implement fit and "
+                                "transform. '%s' (type %s) doesn't" %
+                                (t, type(t)))
+
+    @property
+    def named_transformers_(self):
+        """Access the fitted transformer by name.
+
+        Read-only attribute to access any transformer by given name.
```

Comment:
Should we make sure people can't assign to this? That would probably require a change in ``Bunch``, right?
Someone could try
```python
ct.named_transformers.step_name = Estimator()
```
and it would succeed but do nothing, right?